### PR TITLE
Fix#2057 Crop Feature Added

### LINKED ISSUE #2057: No option to remove profile pic
**Actual Behaviour**

Their is no process to remove profile pic uploaded and neither it does ask the image area that will be uploaded.

**Expected Behaviour**

There should be an option to remove the profile picture uploaded and select the area of the pic that will be uploaded

The profile pic should also be visible in the Side Navigation View aside the name of the logged in user.

**Steps to reproduce it**

Go to Edit Profile
On the Pic, upload any pic 

**Would you like to work on the issue?**

Yes!

@iamareebjamal @heysadboy

### PR DESCRIPTION
**Pull request by codedsun**
Fixes #2057 Crop Feature Added

Changes:
Uploading Image has now crop feature
Added dummy image for profile pic 

Screenshots for the change:

### CODE REVIEW

**Comment by codedsun** [comment]
@the-dagger  Review it plz. 👍

File: android/build.gradle [hunk 1]
```diff
@@ -21,6 +21,10 @@ allprojects {
         maven {
             url 'https://github.com/uPhyca/stetho-realm/raw/master/maven-repo'
         }
+
```
Comment (the-dagger) [inline android/build.gradle]: Why this extra space?
Also can you please add some screenshots? :)

Comment (codedsun) [reply android/build.gradle]: @the-dagger As the instance of openevent is down for current, I will the run the activity directly and will show how the following is working .

**Comment by codedsun** [comment]
screenshot_2017-12-31-02-15-46
screenshot_2017-12-31-02-16-39
screenshot_2017-12-31-02-16-44

@the-dagger Review !

File: android/app/src/main/res/drawable/ic_account_circle_grey_24dp.xml [hunk 2]
```diff
@@ -1,9 +1,5 @@
-<vector xmlns:android="http://schemas.android.com/apk/res/android"
-    android:width="56dp"
-    android:height="56dp"
-    android:viewportHeight="24.0"
-    android:viewportWidth="24.0">
-    <path
-        android:fillColor="#9b9b9b"
-        android:pathData="M12,2C6.48,2 2,6.48 2,12s4.48,10 10,10 10,-4.48 10,-10S17.52,2 12,2zM12,5c1.66,0 3,1.34 3,3s-1.34,3 -3,3 -3,-1.34 -3,-3 1.34,-3 3,-3zM12,19.2c-2.5,0 -4.71,-1.28 -6,-3.22 0.03,-1.99 4,-3.08 6,-3.08 1.99,0 5.97,1.09 6,3.08 -1.29,1.94 -3.5,3.22 -6,3.22z" />
+<vector android:height="24dp" android:viewportHeight="58.0"
```
Comment (iamareebjamal) [inline android/app/src/main/res/drawable/ic_account_circle_grey_24dp.xml]: Don't change the existing vector please, reuse this instead

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: android/app/src/main/res/drawable/ic_account_circle_grey_24dp.xml (hunk 2 above)
Comment (codedsun) [reply android/app/src/main/res/drawable/ic_account_circle_grey_24dp.xml]: @iamareebjamal Do I have to revert to previous vector used and shouldn't used another as stated in the issue?

Comment (codedsun) [reply android/app/src/main/res/drawable/ic_account_circle_grey_24dp.xml]: @iamareebjamal which vector to use now?

Comment (iamareebjamal) [reply android/app/src/main/res/drawable/ic_account_circle_grey_24dp.xml]: The previous one is standard material vector drawable for profile, so let it be the default one

Comment (codedsun) [reply android/app/src/main/res/drawable/ic_account_circle_grey_24dp.xml]: @iamareebjamal Ok! Updating 👍

File: android/app/src/main/res/drawable/ic_account_circle_grey_24dp.xml [hunk 3]
```diff
@@ -7,3 +7,4 @@
         android:fillColor="#9b9b9b"
         android:pathData="M12,2C6.48,2 2,6.48 2,12s4.48,10 10,10 10,-4.48 10,-10S17.52,2 12,2zM12,5c1.66,0 3,1.34 3,3s-1.34,3 -3,3 -3,-1.34 -3,-3 1.34,-3 3,-3zM12,19.2c-2.5,0 -4.71,-1.28 -6,-3.22 0.03,-1.99 4,-3.08 6,-3.08 1.99,0 5.97,1.09 6,3.08 -1.29,1.94 -3.5,3.22 -6,3.22z" />
 </vector>
+
```
Comment (iamareebjamal) [inline android/app/src/main/res/drawable/ic_account_circle_grey_24dp.xml]: Unneeded change

File: android/build.gradle [hunk 4]
```diff
@@ -21,7 +21,11 @@ allprojects {
         maven {
             url 'https://github.com/uPhyca/stetho-realm/raw/master/maven-repo'
         }
-        maven { url "https://jitpack.io" }
+        
+        maven{
+            url "https://jitpack.io"
+        }
+
```
Comment (iamareebjamal) [inline android/build.gradle]: Unneeded change

File: android/app/src/main/java/org/fossasia/openevent/activities/auth/EditProfileActivity.java [hunk 5]
```diff
@@ -230,20 +234,27 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
...
             encodedImage = encodeImage(selectedImage);
 
             OpenEventApp.picassoWithCache
-                    .load(imageUri)
+                    .load(resultUri)
                     .transform(new CircleTransform())
                     .into(avatar);
+        } else if (resultCode == UCrop.RESULT_ERROR) {
+            final Throwable cropError = UCrop.getError(data);
+            Log.e("EditProfileActivity", "UCrop Error"+cropError );
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/auth/EditProfileActivity.java]: Use Timber

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/activities/auth/EditProfileActivity.java [hunk 6]
```diff
@@ -230,20 +234,27 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
...
             encodedImage = encodeImage(selectedImage);
 
             OpenEventApp.picassoWithCache
-                    .load(imageUri)
+                    .load(resultUri)
                     .transform(new CircleTransform())
                     .into(avatar);
+        } else if (resultCode == UCrop.RESULT_ERROR) {
+            final Throwable cropError = UCrop.getError(data);
+            Timber.d("EditProfileActivity", "UCrop Error"+cropError );
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/auth/EditProfileActivity.java]: Spaces around `+` and no space before `)`

File: android/app/src/main/res/drawable/ic_account_circle_grey_24dp.xml [hunk 7]
```diff
@@ -6,4 +6,4 @@
     <path
         android:fillColor="#9b9b9b"
         android:pathData="M12,2C6.48,2 2,6.48 2,12s4.48,10 10,10 10,-4.48 10,-10S17.52,2 12,2zM12,5c1.66,0 3,1.34 3,3s-1.34,3 -3,3 -3,-1.34 -3,-3 1.34,-3 3,-3zM12,19.2c-2.5,0 -4.71,-1.28 -6,-3.22 0.03,-1.99 4,-3.08 6,-3.08 1.99,0 5.97,1.09 6,3.08 -1.29,1.94 -3.5,3.22 -6,3.22z" />
-</vector>
+</vector>
```
Comment (iamareebjamal) [inline android/app/src/main/res/drawable/ic_account_circle_grey_24dp.xml]: Add a newline at end

**Review by iamareebjamal — CHANGES_REQUESTED** [review]

File: android/app/src/main/java/org/fossasia/openevent/activities/auth/EditProfileActivity.java (hunk 6 above)
Comment (codedsun) [reply android/app/src/main/java/org/fossasia/openevent/activities/auth/EditProfileActivity.java]: @iamareebjamal Check gitter plz 👍

**Comment by codedsun** [comment]
@iamareebjamal Resolved all 👍

**Comment by iamareebjamal** [comment]
Just remove the unused Log import and we're done

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**